### DISCUSSION TITLE
How to add body params to the token request query?

### DISCUSSION THREAD

**Discussion by vincaslt** [nextauthjs/next-auth#2775] [Help]
Pinterest API seems to be a bit special as it requires token request params to be provided as body params instead of URL query string: https://developers.pinterest.com/docs/api/v5/#tag/Authentication

From the source code it seems like auth-next only add params to query string. Am I missing something? Is there a way to override this behavior or do I have to fork and fix it?

**Comment by balazsorban44** [comment]
This is not a very special request, you are right wonder. :sweat_smile:

We are trying to help with this and better support similar situations in v4. #1846, #2411

We don't have intentions to backport it to v3. If you want to check it out, you can see the migration docs here: https://next-auth.js.org/getting-started/upgrade-v4

NOTE: v4 is currently in beta!
